StudyPathTraker/StudyPathTraker/Modules/CheckList/New CheckList/SPNewCheckListViewController.swift

```diff
@@ -48,19 +50,26 @@ class SPNewCheckListViewController: UIViewController {
 
     @IBAction private func addNewMilestione(_ sender: Any) {
         let name = nameMilestone.text ?? ""
+        guard let currentItem = item else {
+            return
+        }
         if verifyForm(name: name) {
             if isToEdit {
                 editMilestone(name: name)
             } else {
-                delegate?.didAddNewMilestone(name: name)
+                presenter.addMilestone(createNewMilestone(name: name), item: currentItem)
             }
         }
     }
     private func editMilestone(name: String) {
         guard let currentMilestone = milestone else {
             return
         }
-        delegate?.didEditMilestone(name: name, milestone: currentMilestone)
+        presenter.updateMilestone(currentMilestone, name: name)
+    }
+
+    private func createNewMilestone(name: String) -> Milestone {
```

Comment:
also this, create a extension of Milestone and put this as static function there.